xbmc/cores/AudioEngine/Sinks/pipewire/PipewireNode.h

```diff
@@ -28,21 +28,21 @@ class CPipewireRegistry;
 class CPipewireNode : public CPipewireProxy
 {
 public:
-  explicit CPipewireNode(CPipewireRegistry& registry, uint32_t id, const char* type);
+  CPipewireNode(CPipewireRegistry& registry, uint32_t id, const char* type);
   CPipewireNode() = delete;
   ~CPipewireNode() override;
 
-  void EnumerateFormats();
+  void EnumerateFormats() const;
 
-  pw_node_info* GetInfo() { return m_info.get(); }
+  pw_node_info* GetInfo() const { return m_info.get(); }
 
-  std::set<spa_audio_format>& GetFormats() { return m_formats; }
-  std::set<spa_audio_channel>& GetChannels() { return m_channels; }
-  std::set<uint32_t>& GetRates() { return m_rates; }
-  std::set<spa_audio_iec958_codec>& GetIEC958Codecs() { return m_iec958Codecs; }
+  std::set<spa_audio_format>& GetFormats() const { return m_formats; }
+  std::set<spa_audio_channel>& GetChannels() const { return m_channels; }
+  std::set<uint32_t>& GetRates() const { return m_rates; }
+  std::set<spa_audio_iec958_codec>& GetIEC958Codecs() const { return m_iec958Codecs; }
```

Comment:
`const` function should return `const` reference. non-const function can return non-const reference. For example:

```c++
  const std::set<spa_audio_format>& GetFormats() const { return m_formats; }
  std::set<spa_audio_format>& GetFormats() { return m_formats; }
```

Reply:
I find that when functions are documented, it's better to group by function, e.g. https://github.com/xbmc/xbmc/blob/master/xbmc/games/controllers/types/ControllerNode.h

And when functions are undocumented, it's cleaner to group by constness, e.g. https://github.com/garbear/xbmc/blob/retroengine/xbmc/games/controllers/input/ControllerState.h